src/fheroes2/gui/ui_tool.cpp

```diff
@@ -722,4 +722,43 @@ namespace fheroes2
             ApplyPalette( image, rightRoi.x, rightRoi.y, image, rightRoi.x, rightRoi.y, rightRoi.width, rightRoi.height, paletteId );
         }
     }
+
+    bool processIntegerValueTyping( const int32_t minimum, const int32_t maximum, int32_t & value )
+    {
+        const LocalEvent & le = LocalEvent::Get();
+
+        if ( !le.isAnyKeyPressed() ) {
+            // No key is pressed.
+            return false;
+        }
+
+        if ( le.isKeyPressed( fheroes2::Key::KEY_BACKSPACE ) ) {
+            value = value / 10;
+            return true;
+        }
+
+        if ( minimum < 0 && value != 0 && le.isKeyPressed( fheroes2::Key::KEY_MINUS ) ) {
```

Comment:
Should we also process the keypad minus?
```suggestion
        if ( minimum < 0 && value != 0 && ( le.isKeyPressed( fheroes2::Key::KEY_MINUS ) || le.isKeyPressed( fheroes2::Key::KEY_KP_MINUS ) ) ) {
```

Reply:
It is fixed.